### ISSUE TITLE
Help for terraria and selfhtml wikis

### ISSUE THREAD

**Issue by benoit74** [openzim/mwoffliner#1948]
@VadimKovalenkoSNF, we would probably benefit a lot from your guidance regarding the configuration of these two mwoflliner recipes:
- terrariawiki_en: https://farm.openzim.org/recipes/terrariawiki_en
- selfhtml_de_all: https://farm.openzim.org/recipes/selfhtml_de_all

In both recipes, we get the same error "Unable to find appropriate API end-point to retrieve article HTML".

Note that this is still running on mwoffliner 1.13.0

I had a look with @Popolechien and @RavanJAltaie today and found nothing really weird but I have way too limited knowledge of this scraper.

What I noticed is that both wikis do not have the VisualEditor extension installed:
- https://terraria.wiki.gg/wiki/Special:Version
- https://wiki.selfhtml.org/wiki/Spezial:Version

Luckily terraria seems to be pretty recent (or at least less old than other wikis) and hence has the REST API available, but could this API be used with 1.13.0 or do we have to wait for 1.14.0? 

In other words, is VisualEditor extension a requirement for mwoffliner 1.13.0 ? 

And btw, is there an estimation of 1.14 release schedule?

Thanks!

**Comment by VadimKovalenkoSNF** [comment]
It looks like **terraria.wiki.gg** does not support `action=visualeditor`, check this endpoint https://terraria.wiki.gg/api.php?action=visualeditor&mobileformat=html&format=json&paction=parse&formatversion=2&page=Plants and docs: https://terraria.wiki.gg/api.php --> No visual editor there. For this wiki I would try to use Mediawiki REST API  (`/rest.php`) - https://terraria.wiki.gg/rest.php/v1/page/Plants/html (styling and markup look messy)  or Mediawiki Action API (`action=parse`): https://terraria.wiki.gg/api.php?action=parse&format=json&prop=modules%7Cjsconfigvars%7Cheadhtml%7Ctext&useparsoid=1&page=Plants&formatversion=2

As for **wiki.selfhtml.org** I haven't found working Mediawiki REST API (`/rest.php`) example as well as VE (check https://wiki.selfhtml.org/api.php). Bu Mediawiki Action API (`action=parse`) is working, check: https://wiki.selfhtml.org/api.php?action=parse&format=json&prop=modules%7Cjsconfigvars%7Cheadhtml%7Ctext&useparsoid=1&page=Grundlagen&formatversion=2

Note, that Mediawiki Action API (`action=parse`) is not supported in mwoffliner yet.

> Luckily terraria seems to be pretty recent (or at least less old than other wikis) and hence has the REST API available, but could this API be used with 1.13.0 or do we have to wait for 1.14.0?

Mediawiki REST API (`rest.php`) will be available once https://github.com/openzim/mwoffliner/pull/1947 is approved which is  milestone 1.15.0

> In other words, is VisualEditor extension a requirement for mwoffliner 1.13.0 ?

All recent versions of mwoffliner (1.13.0+) can parse content from VE but only for the main page by default (if no Wikimedia REST API `page/html` is available) but there is no extra checking of **Special:Version** page per wiki from mwoffliner side.

> And btw, is there an estimation of 1.14 release schedule?

^ @kelson42

**Comment by RavanJAltaie** [comment]
https://github.com/openzim/zim-requests/issues/743

**Comment by kelson42** [comment]
I think everything is said: without VE it's not possible to scrape a MW with `1.13.0` and #1601 is not implemented yet.

**Comment by raptor** [comment]
I attempted the Terraria wiki with the latest in git, using the rest API. Here was my command, minimally modified from the recipe:
```
npm run mwoffliner --trace-warnings -- --customZimDescription="The comprehensive reference written and maintained by the players." --customZimFavicon="https://drive.farm.openzim.org/Corrected%20Logos%20for%20recipes/Terraria.png" --customZimTitle="Terraria Wiki" --filenamePrefix="terrariawiki_en" --format="novid:maxi" --mwModulePath="/load.php" --mwMediaWikiRestApiPath="rest.php/v1/page/" --mwUrl="https://terraria.wiki.gg" --mwWikiPath="/wiki/" --osTmpDir="/dev/shm" --outputDirectory="output" --publisher="Kiwix" --verbose
```
It retrieves several pages but fails with a timeout on a rather large page:

   https://terraria.wiki.gg/rest.php/v1/page/List_of_possible_modifiers/html

A re-run just fails with gateway timeouts after this happens.

There are also lots of failed attempts with trying to download different language pages, but this may be due to my lack of skill in using this tool.

**Comment by benoit74** [comment]
Terraria wiki is protected by Cloudflare (see `Server: cloudflare` below), so clearly we have high chances of been blocked by its protections.

```
curl -I terraria.wiki.gg
HTTP/1.1 301 Moved Permanently
Date: Mon, 05 Feb 2024 08:01:33 GMT
Connection: keep-alive
Cache-Control: max-age=3600
Expires: Mon, 05 Feb 2024 09:01:33 GMT
Location: https://terraria.wiki.gg/
Server: cloudflare
CF-RAY: 8509a04a18c74e1c-HEL
alt-svc: h3=":443"; ma=86400
```

**Comment by Popolechien** [comment]
We need to define a Cloudflare policy of sorts, if only to rejects demands outright, or explain to requesters how to whitelist us.
